plugin/file/lookup.go

```diff
@@ -62,6 +61,11 @@ func (z *Zone) Lookup(ctx context.Context, state request.Request, qname string)
 		elem, wildElem *tree.Elem
 	)
 
+	loop := 0
+	if x := ctx.Value(loopKey); x != nil {
```

Comment:
Do we need to be concerned about leaking the loop data to other plugins in the chain?

Reply:
as far as I understand context.Context you can't access this because `loopKey` is private.

Reply:
OTOH having the ability for other plugins (or the server?) to check this might be a possibility

Reply:
maybe doing this more generally is actually a good thing. Also returning SERVFAIL might be an option here, and not a garbled response